src/envoy/http/authn_wasm/config.cc

```diff
@@ -0,0 +1,113 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "src/envoy/http/authn_wasm/config.h"
+
+#include "authentication/v1alpha1/policy.pb.h"
+#include "google/protobuf/text_format.h"
+#include "google/protobuf/util/json_util.h"
+
+namespace iaapi = istio::authentication::v1alpha1;
+
+namespace Envoy {
+namespace Extensions {
+namespace Wasm {
+namespace AuthnWasm {
+
+// imports from the low-level API
+using Common::Wasm::Null::NullVmPluginFactory;
+using Common::Wasm::Null::Plugin::logDebug;
+using Common::Wasm::Null::Plugin::logError;
+
+using ::google::protobuf::TextFormat;
+using namespace google::protobuf::util;
+
+void PluginRootContext::onConfigure(std::unique_ptr<WasmData> configuration) {
+  logDebug("called PluginRootContext::onConfigure()");
```

Comment:
different filter will share the same function exported by host? maybe we should then add 'authn' prefix to simulate log scope?

Reply:
That's a good point but I think it should be solved in the WASM layer, so I will just add a TODO at this moment.

Reply:
Have you added TODO here? didn't see the diff yet.

Reply:
never mind see it above.